ext/search_kit/Civi/Api4/Action/SearchDisplay/AbstractRunAction.php

```diff
@@ -451,6 +451,16 @@ protected function getRuleCondition($clause, bool $isMulti = FALSE): array {
       }
       $fieldKey = $this->renameIfAggregate($fieldKey);
     }
+    // Handle relative dates
+    if (!empty($clause[2])) {
+      $field = $this->getField($fieldKey);
+      if ($field && $field['data_type'] === 'Date') {
+        $clause[2] = date('Y-m-d', strtotime($clause[2]));
```

Comment:
seems will still fail for relative dates?  eg

![image](https://github.com/civicrm/civicrm-core/assets/5929648/acc7fa16-11d0-49c1-af2a-6e7683a0b77a)

where $clause[2] is `this.day`. Might require modification to use CRM_Utils_Date helpers instead?

Reply:
I think the `Date Range` option doesn't make sense with operators other then `Is between`. Suppose a end-user chose >= with previous calendar year then it translate the date range between 1-1-2023 to 31-12-2023 and thus any operator other than `Is between` isn't applicable. So I would say date range option should be enabled when `Is between` is selected. And when `Is between` is selected, like in Civi Reports date filter, there should be date range option and relative date options. Currently it shows From and To datepickers.

Reply:
This PR didn't get ambitious enough to support date ranges. Could do so as a followup but imo this pr is better in than out.